### ISSUE TITLE
PF4 Spike: How to handle animations? 

### ISSUE THREAD

**Issue by LHinson** [patternfly/patternfly-react#689]
1. Work with team to Identify the process for how we will convey the intended animations from design to react
2. Define how we will handle animations in PF-React
3. Create an issue to update nav with animations once the technical direction is identified
4. Work with @kybaker to understand if any React components are missing animations that should be there. Create issues for each of these if so. 

Required stakeholder for discussion: 
Design and Visuals: @mcarrano @kybaker @matthewcarleton @mceledonia

**Comment by karelhala** [comment]
I'd like to add another point to this, it might be just optional, but might be good to track it down - icon transition animations. We didn't used it much in PF3 but I think that it adds some nice effect to icons and since we have let's say icons for different state of resource capacity, lock/unlock and some more we might consider this to add this as well. I love [Morpheus](https://github.com/alexk111/SVG-Morpheus), but that library is no longer maintained, but we have plenty of other options like using [svg.js](http://svgjs.com/) which looks like the most favorite and probably most used library for animations or another option is to use [kute.js](http://thednp.github.io/kute.js/svg.html), small library which focuses mostly on just animations of svg.

**Comment by mturley** [comment]
I started attempting to work on animation of the expandable nav in https://github.com/patternfly/patternfly-react/pull/819: currently that implementation is broken, but it sparked a little bit of a discussion with @priley86 and @dmiller9911.

So far, what I've gathered is that there are a lot of libraries out there for doing declarative animations in React, which directly apply styles to the DOM (https://logrocket.com/blog/comparing-the-top-js-animation-libraries-for-use-in-react-apps-2/). But I think that since these animations have already been defined in PF4 core as CSS transitions, we risk inconsistency between core and PF-React if we go with one of those libraries and implement the animations again. Maybe there will be exceptions to this, though.

What I'm leaning towards is using something like https://reactcommunity.org/react-transition-group/transition to imitate the JS from the core examples as closely as possible, just applying attributes and CSS classes with some built in delays.
